src/main/java/com/iridium/iridiumskyblock/gui/BlockValueGUI.java

```diff
@@ -84,9 +84,18 @@ public void onInventoryClick(InventoryClickEvent event) {
         // Don't do anything here, it gets cancelled automatically
     }
 
-    enum BlockValueType {
+    public enum BlockValueType {
         BLOCK,
-        SPAWNER
+        SPAWNER;
+
+        public static BlockValueType getType(String type) {
```

Comment:
Just use Enum#valueOf()?

Reply:
Edit: You probably want to overwrite Enum#valueOf() and use streams like in other enums

Reply:
few issues, 1 this is case sensitive and 2 this will throw an error if not found. Mabey im being dumb and there is an easy way to fix both of these but I cant think of one other than doing this

Reply:
You did it better in the MissionType GUI, just rename this method to valueOf() so it overwrites the default method (+ add @Override)